skglm/experimental/smooth_quantile_regressor.py

```diff
@@ -0,0 +1,553 @@
+        - 'small_problem_threshold': int, default=100
+          Threshold for considering a problem "small" in terms of features.
+        - 'p0_frac_large': float, default=0.1
+          Fraction of features to use in working set for large problems.
+        - 'p0_frac_small': float, default=0.5
+          Fraction of features to use in working set for small problems.
+        - 'p0_min': int, default=10
+          Minimum working set size for any problem.
+
+    Attributes
+    ----------
+    coef_ : ndarray of shape (n_features,)
+        Parameter vector (:math:`w` in the cost function formula).
+
+    intercept_ : float
+        Intercept term in decision function.
+
+    stage_results_ : list
+        Information about each smoothing stage including:
+        - delta: smoothing parameter
+        - obj_value: objective value
+        - coef_norm: L2 norm of coefficients
+        - quantile_error: absolute difference between target and achieved quantile
+        - actual_quantile: achieved quantile level
+        - coef: coefficient vector
+        - intercept: intercept value
+        - n_iter: number of iterations (if available)
+
+    Notes
+    -----
+    This implementation uses a progressive smoothing approach to solve quantile
+    regression problems. It starts with a highly smoothed approximation and
+    gradually reduces the smoothing parameter to approach the original non-smooth
+    problem. This approach is particularly effective for large datasets where
+    direct optimization of the non-smooth objective can be challenging.
+
+    The solver automatically selects the best solution from all smoothing stages
+    based on the quantile error, ensuring good approximation of the target
+    quantile level.
+
+    References
+    ----------
+    Chen, C. (2007). A Finite Smoothing Algorithm for Quantile Regression.
+    Journal of Computational and Graphical Statistics, 16(1), 136–164.
+    http://www.jstor.org/stable/27594233
+
+    Examples
+    --------
+    >>> from skglm.experimental.smooth_quantile_regressor import SmoothQuantileRegressor
+    >>> import numpy as np
+    >>> X = np.random.randn(1000, 10)
+    >>> y = np.random.randn(1000)
+    >>> reg = SmoothQuantileRegressor(quantile=0.8, alpha=0.1)
+    >>> reg.fit(X, y)
+    >>> print(reg.coef_)
+    """
+
+    def __init__(
+        self,
+        smoothing_sequence=None,
+        quantile=0.5,
+        alpha=0.1,
+        initial_delta=0.5,
+        min_delta=1e-6,
+        smooth_solver=None,
+        verbose=False,
+        delta_tol=1e-6,
+        max_stages=8,
+        quantile_error_threshold=0.005,
+        solver_params=None,
+        initial_alpha: float = None,
+        alpha_schedule: str = 'geometric',
+    ):
+        self.quantile = float(quantile)
+        if not 0 < self.quantile < 1:
+            raise ValueError("quantile must be between 0 and 1")
+        self.alpha = float(alpha)
```

Comment:
look at other parts of the code,we do not cast stuff as float (a priori it's not needed, python will do it itself if need be)